x/feemarket/keeper/feemarket.go

```
@@ -22,7 +23,7 @@ func (k *Keeper) UpdateFeeMarket(ctx sdk.Context) error {
 
 	state, err := k.GetState(ctx)
 	if err != nil {
-		return err
+		panic(err)
```

If we panic here the chain will halt right? Shouldn't we return the error without applying the changes and let the parent decide if it should panic?

---

Im thinking of moving the panic to one level up at GetState. My logic is that getState should never error and if it does there is something seriously wrong and chain should halt. I like to only use panic at places where it should never error.

---

I see your point, it makes sense